inst/@sym/assume.m

```diff
@@ -116,47 +116,48 @@
 %% @seealso{@@sym/assumeAlso, assumptions, sym, syms}
 %% @end deftypemethod
 
-%% Author: Colin B. Macdonald
-%% Keywords: symbolic
 
-function varargout = assume(x, varargin)
+function varargout = assume(xx, varargin)
 
   assert (nargin > 1, 'General algebraic assumptions are not supported');
 
-  xstr = x.flat;
+  for i = 1:length (xx)
+    x = subsref (xx, substruct('()', {i}));
+    xstr = x.flat;
 
-  if (nargin > 1 && strcmp(varargin{1}, 'clear'))
-    assert (nargin == 2, 'assume: clear cannot be combined with other assumptions')
-    newx = sym(xstr);
-  else
-    for n = 2:nargin
-      cond = varargin{n-1};
-      ca.(cond) = true;
+    if (nargin > 1 && strcmp(varargin{1}, 'clear'))
+      assert (nargin == 2, 'assume: clear cannot be combined with other assumptions')
+      newx = sym(xstr);
+    else
+      for n = 2:nargin
+        cond = varargin{n-1};
+        ca.(cond) = true;
+      end
+      newx = sym(xstr, ca);
     end
-    newx = sym(xstr, ca);
-  end
 
-  if (nargout > 0)
-    varargout{1} = newx;
-    return
-  end
+    if (nargout > 0)
+      varargout{i} = newx;
```

Comment:
`f` should not be updated.  `assumptions(f)` should be empty.  I believe this is tested.

I don't think the return is missing: the "mucking around" bit is in a "else" block.